### ISSUE TITLE
2018 releases will not boot on RPi 3 B+

### ISSUE THREAD

**Issue by Nilpo** [maxnet/berryboot#495]
BerryBoot will not boot on Raspberry Pi 3 B+. The power LED (red) is lit, but the ACT LED is not. Typically this would indicate a bad microSD card or boot image. The Pi never loads the GPU screen test (rainbow).

I have bought multiple boards with multiple microSD cards of several sizes and brands. All equipment is known to be working. All cards and boards will boot both NOOBS and Raspbian images.

I tried older versions of BerryBoot and had the same results until I tried `berryboot-20170527-pi2-pi3` which boots to the rainbow with a lightning bolt indicating outdated firmware.

Armed with this information, I reformatted and loaded the latest berryboot release again and tested by replacing files from the "working" version. What I've found is that I can use the latest release and replace `start_cd.elf`, `start_x.elf`, and `start.elf` from the 20170527 release and achieve the same partial boot indicating outdated firmware. However, using any of those three files from 2018 releases results in no boot at all.

I can't find any documentation for this, but perhaps it will help someone. When I get partial boot, the green ACT LED is steady on and the red PWR LED flashes 4 times.

**Comment by maxnet** [comment]
>The Pi never loads the GPU screen test (rainbow).

The rainbow screen is drawn by the closed source Raspberry Pi firmware files that we did not create, so that does not sound like a Berryboot specific problem.


I suggest you try replacing files with different versions from: https://github.com/raspberrypi/firmware/commits/master/boot/start_x.elf
(click on the "<>" button to browse the repository at a point in time, and then "download")
and find the exact firmware release where things break.

The default config.txt Berryboot has uses the _x version of the firmware files.
So replace at least `bootcode.bin`, `start_x.elf` and `fixup_x.dat` each time.

Once you find exact commit, please report the issue to the firmware repo.

**Comment by symbios24** [comment]
check another power supply this is also a supply issue

**Comment by alfieoriginal** [comment]
I agree, power supply or power cable issue

**Comment by Nilpo** [comment]
Again, I can successfully but everything except BerryBoot and I have used multiple known good power supplies. This is not a power supply issue.

**Comment by macmpi** [comment]
> which boots to the rainbow with a **lightning bolt** indicating...

Lightning bolt does indicate insufficient power actually.
In your other distributions you may have some kernel messages about this too (if running on 4.14):
`dmesg  | grep voltage`
They may still boot fine, but problems/instabilities may arise.

Not sure latest stock [4.14.34](https://github.com/Hexxeh/rpi-firmware/commits/master) kernel resolves all issues on 3b+ yet.

**Comment by Nilpo** [comment]
> Lightning bolt does indicate insufficient power actually. In your other distributions you may have some kernel messages about this too (if running on 4.14)

Yes, but that is not it's only meaning. On the RPi 3 B+, the lighting bolt on the GPU test screen indicates outdated firmware.

I've already ruled out power as an issue.

**Comment by symbios24** [comment]
Did you use the same power cable with all the supplies you tested? Rpi3B+ uses almost twice power as Rpi3B so its recommended that you have a good power supply like the official one(2,5a)

**Comment by symbios24** [comment]
Did you use the same power cable with all the supplies you tested? Rpi3B+ draws almost twice as power compared to Rpi3B so its recommended that you have a good power supply like the official one(2,5a)

**Comment by symbios24** [comment]
Did you use the same power cable with all the supplies you tested? Rpi3B+ draws almost twice as power in some cases compared to Rpi3B so its recommended that you have a good power supply like the official one(2,5a)

**Comment by macmpi** [comment]
You may want to try newer firmware files as explained earlier by @maxnet and report.

In particular yesterday firmware release sets a more conservative SDRAM clock-rate, as it seems [some boards may have issues](https://www.raspberrypi.org/forums/viewtopic.php?f=28&t=208821&start=150#p1303015).

**Comment by Nilpo** [comment]
Two of them are official RPF power supplies for the Pi 3 (the 2.5A ones). Another is a quality third-party supply that is 5V3A. The other two are larger supplies. One is 5.1V3A (for the WD PiDrive), the other is 5V5A for running a Pi enclosure with external HDD. All of them have their own built in cords so there were no re-used parts during those tests.

For each of those tests I used the same microSD card to boot NOOBS, Raspbian, and BerryBoot on two different RPi 3 B+. In all cases, I can boot everything except BB. I then tried the same tests with several other known good microSD cards all with the same results. I used the latest official release version of each. (I'm using the official SD Card Formatter on both Windows and Mac, Etcher for writing images, and the rest of the cards were made manually.)

I've effectively ruled out power, microSD cards, and formatting as potential issues.

I tried the latest firmware (at the time) with BB and could not boot. I can not boot any version of BB from 2018 so far. I have not tried with yesterday's firmware. I will try that and report back.

I also understand that you can watch boot messages over serial, but I have not hooked it up to do that yet. I may make some time tomorrow for that.

**Comment by maxnet** [comment]
>You may want to try newer firmware files as explained earlier by @maxnet and report.

Actually, I suggested trying OLDER files until finding a release that works.
As he mentioned it did work in Raspbian, and we have newer files than what ships with that.

**Comment by symbios24** [comment]
i have the feeling Rpi3B+ is problematic high temps/high current/compatibility problems etc.
10% speed boost is a joke.300mbps lan another joke 5ghz wifi nobody use it bluetooth 4.2 the same,octacore and uhs-1 card speed is what we need not the same over and over

**Comment by maxnet** [comment]
>i have the feeling Rpi3B+ is problematic

Well, not all 3+s have this problem.
The two I have do run fine.

**Comment by Nilpo** [comment]
I spent the day (literally) doing extensive testing. For testing purposes, I used the same microSD card for all tests. I randomly duplicated some of the tests throughout with another microSD card and had the same results. To avoid the risk of power issues, I used a 5V3A power supply.

I tested every version of BerryBoot from the last 2016 release until present. For each release, I tested the release firmware and each firmware from the last 2016 forward. As a comparison, I tested every release of NOOBS for the same time period. Here are the tested softwares.

 - BerryBoot

     - berryboot-20161118-pi2-pi3
     - berryboot-20170208-pi2-pi3
     - berryboot-20170308-pi2-pi3
     - berryboot-20170509-pi2-pi3
     - berryboot-20170527-pi2-pi3
     - berryboot-20180216-pi2-pi3
     - berryboot-20180314-pi2-pi3
     - berryboot-20180322-pi2-pi3
     - berryboot-20180329-pi2-pi3
     - berryboot-20180405-pi2-pi3
     - berryboot-20180405-pi0-pi1-pi2-pi3

 - NOOBS

     - NOOBS-2016-11-29 (v2.1.0)
     - NOOBS-2017-02-27 (v2.2.0)
     - NOOBS-2017-03-03 (v2.3.0)
     - NOOBS-2017-04-10 (v2.4.0)
     - NOOBS-2017-06-23 (v2.4.1)
     - NOOBS-2017-07-05 (v2.4.2)
     - NOOBS-2017-08-17 (v2.4.3)
     - NOOBS-2017-09-08 (v2.4.4)
     - NOOBS-2017-12-01 (v2.4.5)
     - NOOBS-2018-03-14 (v2.7.0)

 - Firmwares

     - firmware-1.20161215
     - firmware-1.20170215
     - firmware-1.20170303
     - firmware-1.20170405
     - firmware-1.20170427
     - firmware-1.20170515
     - firmware-1.20170702
     - firmware-1.20170703
     - firmware-1.20170811
     - firmware-1.20171029
     - firmware-1.20180313
     - firmware-1.20180328
     - firmware-master (a154f2136850dba827cf4bc40794854376902cbd 4-17-2018)

Here are my results.

### BerryBoot

I was unable to boot any version of BerryBoot with any version of firmware. By unable to boot, I mean that I never successfully reached the GUI test screen.

 - All 2018 versions of BerryBoot:

    - 2018 firmwares - No boot, solid RED, no GREEN
    - 2017 firmwares - No boot, no RED, 4 long GREEN, 4 short GREEN

 - All 2017 versions of BerryBoot:

    - released version - Boot to GUI with lightning bolt, 4 long RED, 4 short RED, no GREEN
    - with later 2017 firmwares - No boot, no RED, 4 long GREEN, 4 short GREEN
    - 2018 firmwares - No boot, solid RED, no GREEN

 - Last 2016 release was the closest success:

    - boots to GUI test with lightning bolt, 4 long RED, 4 short RED, no GREEN

### NOOBS

I was able to boot every version of NOOBS to the GUI test screen with a lightning bolt.

 - Every version: boots to GUI test with lightning bolt. 2016 version has 4 long RED, 4 short RED, no GREEN. All newer versions have no RED, 4 long GREEN, 4 short GREEN.
 - The latest release v2.7.0 boots perfectly and runs as expected. It fully installs and has been running flawlessly for several days.

In summary, I have not successfully booted any version of BerryBoot on a Raspberry Pi 3 B+. Every version of NOOBS will at least boot to the GUI test. The most recent version of the NOOBS is the only that will actually fully boot to the OS.

The important takeaway for me is that even on the same firmware, BerryBoot will not boot as far as NOOBS in any instance.

**Comment by symbios24** [comment]
better return your rpi3b+ for rma its no use to keep trying. its faulty

**Comment by Nilpo** [comment]
I've already replaced both of them. I'm on my third and fourth.

(They are all from the same supplier though.)

I wonder if they have a bad lot of them.

**Comment by symbios24** [comment]
probably,take your money back if possible and buy from another supplier.or keep your money in my opinion and keep using your rpi3b if you have one.rpi3b+ its not really an update one and the same.

**Comment by maxnet** [comment]
Does it work better if in config.txt you remove:

```
disable_overscan=1
start_x=1
```

And put in HDMI options like:

```
hdmi_safe=1
```

**Comment by Nilpo** [comment]
> Does it work better if in config.txt you remove:
> 
>     disable_overscan=1
>     start_x=1
> 
> And put in HDMI options like:
> 
>     hdmi_safe=1

Yes! It boot perfectly.

**Comment by maxnet** [comment]
And with `hdmi_force_hotplug=1` instead of `hdmi_safe=1` ?
If it also works with that, you have a broken HDMI cable, that is missing a wire.

**Comment by Nilpo** [comment]
That also works. (I'm testing on the latest release 2018-04-05) The graphics are huge on screen, but it does boot.

I have some brand new high quality cables in inventory. I'll test with different cables later today and report back.

While I'm at it, I will test on different displays. I'm currently going through an active HDMI to DVI adapter. It shouldn't have any effect, but I will find some displays with pure HDMI input.

As luck would have it, I also found a microSD card with BerryBoot from many years ago from either a Pi 2 or right after I bought my first Pi 3. Interestingly, it booted immediately.

**Comment by maxnet** [comment]
HDMI-to-DVI should be passive.
Same signal, different connector.

**Comment by Nilpo** [comment]
Okay, the problem was, in fact, related to the HDMI. I was plugged to an HDMI -> VGA adapter. I tried using an HDMI -> DVI-I adapter as well as an HDMI display and both worked. Every 2018 release boots perfectly. 2017 and earlier releases boot to GUI with lightning bolt indicating outdated firmware as expected. For the sake of testing, I used the same HDMI cable that I've used all along.

The problem either lies in the VGA adapter (which I've used with my Raspberry Pis for several years now) or it is something specific to the HDMI output range or resolution of the Pi 3 B+.

I recall now why I was using the VGA adapter instead of the DVI. It provides an audio breakout that allows me to have sound. Several of the cases I use block access to the 3.5mm audio jack so I lose audio when using DVI. But I suppose that's a sacrifice I'm willing to make.

**Comment by maxnet** [comment]
>or it is something specific to the HDMI output range or resolution of the Pi 3 B+.

Could also be something more subtle.
E.g. that the 3+ is slightly faster, and isn't waiting long enough for your active adapter to get ready, and falls back to quickly to analog video.

Can try running the 3+ at lower clockspeed with config.txt options like `arm_freq=1200` (that's what 3 without plus uses)

Edit: or play with options like `bootcode_delay=1`

**Comment by Nilpo** [comment]
Changing `arm_freq` had no effect, but adding `bootcode_delay=1` did the job. It also fixes the resolution problem. I think you're right about the active adapter not being ready soon enough. That's a good find. I honestly never even thought about it.

**Comment by Nilpo** [comment]
I think we've successfully determined the cause. I'm going to close this issue now. Thanks for everyone's help.

**Comment by MrJacz** [comment]
I'm having the same problem, its a RaspberryPI 3 model b+. Red light stays on but no GPU test screen or BerryBoot GUI. I've tested with 2-3 power supplies. I use the official power supply. The PI boots NOOBS and Rasbian just fine with no issues. The SD Card was purchased on the 26 of may so couple weeks ago same with the rPI. I'm a bit of a noob and would just like to be able to install multiple OSes whenever so this seems like a helpful I'd like to get working.

Update: The entire problem was due to the fact that my 64gb SD card wasn't formatted to fat32 bc of windows, finally worked it out. Maybe you could add a like common problems section on the readme for people like me? 😉

**Comment by symbios24** [comment]
Nilpo what exactly did you change in config.txt file?
 I`m having the huge graphics on screen.

**Comment by Nilpo** [comment]
In the end, I did not have to change anything. I had to ensure that I was using a quality HDMI cable and a monitor with HDMI input.

`bootcode_delay=1` did proof otherwise useful though.
